src/solvers/strings/string_constraint_generator_main.cpp

```diff
@@ -441,9 +441,12 @@ exprt zero_if_negative(const exprt &expr)
 /// the return codes and merging the constraints.
 std::pair<exprt, string_constraintst> combine_results(
   std::pair<exprt, string_constraintst> result1,
-  std::pair<exprt, string_constraintst> result2)
+  std::pair<exprt, string_constraintst> result2,
+  const namespacet &ns)
 {
-  const exprt return_code = maximum(result1.first, result2.first);
+  exprt return_code = maximum(result1.first, result2.first);
   merge(result2.second, std::move(result1.second));
+  simplify(return_code, ns);
+
   return {return_code, std::move(result2.second)};
```

Comment:
May I suggest to use `return {simplify_expr(return_code, ns), std::move(result2.second)};` at which point the `return_code` can be made `const exprt`.

Reply:
:+1: